test/index.js

```diff
@@ -40,24 +171,62 @@ test('discover and connect', async ({pass, same}) => {
   await client.bind()
   const peer = await client.lookupOne(topic)
   const socket = await client.connect(peer)
-  socket.write('a')
+  socket.write('from client')
   const [ data ] = await once(socket, 'data')
-  same(data, Buffer.from('b'))
-  await close([ client, network ])
+  is(data.toString(), 'from network')
+  await network.close()
+  await client.close()
+})
+
+test('two networks, connect and close', async ({ ok, pass }) => {
+  const network = promisifyApi(guts({
+    socket (sock) { ok(sock, 'got server socket') }
+  }))
+  await network.bind()
+  const client = promisifyApi(guts())
+  const { port } = network.address()
+  const host = '127.0.0.1'
+  const socket = await client.connect({ port, host })
+  ok(socket, 'got client socket')
+  await client.close()
+  pass('client closed')
+  await network.close()
+  pass('network closed')
 })
 
-test('2 networks, connect and close', async ({ ok, pass }) => {
+test('referrer node (remote peer)', async ({ ok, pass }) => {
```

Comment:
forgot to finish this test, ignore for now – it doesn't currently actually test referrer but will come back to this in a separate PR